com.ca.lsp.cobol/lsp-service-cobol/src/test/java/com/ca/lsp/cobol/service/delegates/HighlightsTest.java

```diff
@@ -0,0 +1,105 @@
+ *   Broadcom, Inc. - initial API and implementation
+ */
+
+package com.ca.lsp.cobol.service.delegates;
+
+import static com.ca.lsp.cobol.usecases.UseCaseUtils.createServer;
+import static com.ca.lsp.cobol.usecases.UseCaseUtils.runTextValidation;
+import static com.ca.lsp.cobol.usecases.UseCaseUtils.waitForDiagnostics;
+
+import com.ca.lsp.cobol.service.mocks.TestLanguageClient;
+import com.ca.lsp.cobol.usecases.UseCaseUtils;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.concurrent.ExecutionException;
+import java.util.stream.Collectors;
+import org.eclipse.lsp4j.*;
+import org.eclipse.lsp4j.services.TextDocumentService;
+import org.junit.Before;
+import org.junit.Test;
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertTrue;
+
+public class HighlightsTest {
+  private static final String TEXT =
+      "       Identification Division. \n"
+          + "       Program-id.    ProgramId.\n"
+          + "       Data Division.\n"
+          + "       Working-Storage Section.\n"
+          + "       01   outer1. \n"
+          + "        02   inner1      PIC 9(4) Binary. \n"
+          + "        02   inner2      PIC X(10).\n"
+          + "       01   outer2.\n"
+          + "        02   inner1      PIC 9(4) Binary. \n"
+          + "        02   inner2      PIC X(10).\n"
+          + "       02   Str          PIC 9(4) Binary. \n"
+          + "       Procedure Division.\n"
+          + "       000-Main-Logic.\n"
+          + "           Perform 100-Test.\n"
+          + "           Stop Run.\n"
+          + "       100-Test.\n"
+          + "           Move inner1 of outer1 to Str.\n"
+          + "           Move inner2 of outer1 to Str.\n"
+          + "           Move inner1 of outer2 to Str.\n"
+          + "           Move inner2 of outer2 to Str.\n"
+          + "       End program ProgramId.";
+
+  /*
+   * Testing Document Highlighting with a variable, position is passed in INNER1_POSITION
+   * and the EXPECTED_RANGES is where the expected highlighting is to be enabled
+   */
+
+  private static final Position INNER1_POSITION = new Position(16, 18);
+
+  private static final List<Range> EXPECTED_RANGES = new ArrayList<>();
+
+  static {
+    EXPECTED_RANGES.add(new Range(new Position(5, 13), new Position(5, 19)));
+    EXPECTED_RANGES.add(new Range(new Position(8, 13), new Position(8, 19)));
+    EXPECTED_RANGES.add(new Range(new Position(16, 16), new Position(16, 22)));
+    EXPECTED_RANGES.add(new Range(new Position(18, 16), new Position(18, 22)));
+  }
+
+  private TextDocumentService service;
+
+  /*
+   * Initialising the server
+   */
+
+  @Before
+  public void initializeService() {
+    TestLanguageClient client = new TestLanguageClient();
+    service = createServer(client);
+    runTextValidation(service, TEXT);
+    waitForDiagnostics(client);
+  }
+
+  /*
+   * Testing if Range from Highlights function is providing proper ranges against EXPECTED_RANGES
```

Comment:
This comment should be a part of the test javadoc

Reply:
Correction applied